src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -171,59 +170,153 @@ class IpexCommunicationService extends AgentService {
   }
 
   @OnlineOnly
-  async offerAcdcFromApply(notification: KeriaNotification, acdc: any) {
-    const msgSaid = notification.a.d as string;
-    const msg = await this.props.signifyClient.exchanges().get(msgSaid);
+  async offerAcdcFromApply(id: string, acdc?: any) {
+    const applyNoteRecord = await this.notificationStorage.findById(id);
+
+    if (!applyNoteRecord) {
+      throw new Error(
+        `${IpexCommunicationService.NOTIFICATION_NOT_FOUND} ${id}`
+      );
+    }
+
+    if (Object.keys(applyNoteRecord.linkedGroupRequests).length) {
+      for (const said of Object.keys(applyNoteRecord.linkedGroupRequests)) {
+        if (!applyNoteRecord.linkedGroupRequests[said]) {
+          await this.offerAcdcFromMultisigExn(said as string);
+        }
+      }
+      return;
+    }
 
+    const msgSaid = applyNoteRecord.a.d as string;
+    const applyExn = await this.props.signifyClient.exchanges().get(msgSaid);
+    const holder = await this.identifierStorage.getIdentifierMetadata(
+      applyExn.exn.a.i
+    );
     const holderSignifyName = (
```

Comment:
`discloserSignifyName` but this can also be changed after Martin's PR merges tomorrow